src/pages/Login/login.tsx

```diff
@@ -0,0 +1,90 @@
+import { useState } from 'react';
+import { useForm, SubmitHandler } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import Cookies from 'js-cookie';
+import { Link, useNavigate } from 'react-router-dom';
+import { AiFillExclamationCircle} from 'react-icons/ai';
+import showToast from 'utils/toastMessage';
+import { loginSchema } from 'utils/validationSchema';
+import { IApiResponse, ILogin } from 'utils/interface';
+import { loginUser } from 'api/user';
+import { CONSTANTS } from 'utils/constant';
+import './style.css';
+
+function Login () {
+  const navigate = useNavigate();
+  const { register, handleSubmit, formState: { errors } } = useForm<ILogin>({
+    resolver: yupResolver(loginSchema),
+    mode: 'onBlur',
+  });
+
+  const [showPassword, setShowPassword] = useState(false);
+
+  const togglePasswordVisibility = () => setShowPassword(!showPassword);
+
+  const onSubmit: SubmitHandler<ILogin> = async (data) => {
+  try {
+    const response : IApiResponse = await loginUser(data);
+    const token = response.data;
+    const expirationTime = CONSTANTS.TOKEN_EXPIRE;
+    Cookies.set('token', token, { expires: new Date(Date.now() + expirationTime) });
+    if (response.success) {
+      showToast(response);
+      navigate('/dashboard');
+    }
+  } catch (error: IApiResponse | any) {
+    showToast(error);
+  }
```

Comment:
all of this code will be moved to saga. right? here only an action will be called.